h2o-automl/src/main/java/ai/h2o/automl/Models.java

```diff
@@ -0,0 +1,64 @@
+package ai.h2o.automl;
+
+import hex.Model;
+import hex.ModelContainer;
+import water.*;
+import water.util.ArrayUtils;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
+public class Models<M extends Model> extends Lockable<Models<M>> implements ModelContainer<M> {
+
+    private final Class<M> _clz;
```

Comment:
we usually avoid serializing the Class objects, instead we just pass around typeid - see eg. the constructor of the Job class for an example